File: java/src/main/java/com/mlt/cli/Timer.java
package com.mlt.tools;

public class Timer {
    private long startTime;

    public Timer() {
        startTime = System.nanoTime();
    }

    public void restart() {
        startTime = System.nanoTime();
    }

    public void stop(String message) {
        long endTime = System.nanoTime();
        long elapsedTime = (endTime - startTime) / 1000000; // divide by 1000000 to get milliseconds
        System.out.println("Time elapsed for " + message + ": " + elapsedTime + " milliseconds");
    }
}

File: java/mlt-cli/src/main/kotlin/org/maplibre/mlt/cli/Timer.kt
package org.maplibre.mlt.cli

import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Timer {
    private var startTime: Long

    init {
        startTime = System.nanoTime()
    }

    fun restart() {
        startTime = System.nanoTime()
    }

    fun stop(message: String?) {
        val endTime = System.nanoTime()
        val elapsedTime = (endTime - startTime) / 1_000_000 // divide by 1000000 to get milliseconds
        logger.info("Time elapsed for {}: {} ms", message, elapsedTime)
    }

    private val logger: Logger = LoggerFactory.getLogger(Timer::class.java)
}
